cds-snc/covid-alert-app
# Fix scrolling menu

# Summary

When in landscape mode, some screens (specifically the No Exposure screen, as pointed out in https://github.com/cds-snc/covid-alert-app/issues/955) are clipped at the bottom by the overlay menu.

This is caused by the decision to have the `ScrollView` behind the `OverlayView`. This is a nice visual effect to let the content 'scroll behind' in a nice way. This could be implemented with `contentInsets` and without the weird gap, but a simpler fix here is to simply ensure that there is padding.

In the below screenshot, you see that the contentView (aka the content) only extends a portion of the screen, but the scroll view (blue) sits below only a portion of the menu. In a future PR that should probably be fixed / cleaned up but is very low priority.

<img width="445" alt="Screen Shot 2020-07-30 at 6 23 35 PM" src="https://user-images.githubusercontent.com/128299/88980826-83695d00-d292-11ea-9e59-7db18e5b7037.png">

Additionally, this restores "proper scrolling" and bounciness to the `ScrollView` if and only if the content is longer than the scrollable area. This looks and feels better.

# Test instructions

The test here is to ensure that all screens are visible in both portrait and in landscape mode. There should be _zero_ content that cannot be reached.

Additionally, it should be tested that when in portrait, the screens are not scrollable UNLESS there is content below the menu.

Since this affects the `BaseHomeView` the following screens need to be viewed, and tested:
- No Exposure screens (No region, region covered, region not covered)
- Bluetooth Disabled
- Exposure Notifications disabled
- Framework Unavailable
- Network Disabled
- Exposed
- Diagnosed / Share
- Unknown Problem View

You should see behaviour like this on most pages.

**The view will be static. You cannot scroll this as there is not enough content!**
<img width="434" alt="Screen Shot 2020-07-30 at 6 36 31 PM" src="https://user-images.githubusercontent.com/128299/88981296-a1838d00-d293-11ea-989f-26d3780de96f.png">

**In landscape, you can scroll and it bounces.**
Please note that the colour changes in the menu was caused by the gif capture!
![bounce](https://user-images.githubusercontent.com/128299/88981407-dc85c080-d293-11ea-8682-63c9d2483cd0.gif)

# Help requested

I was able to test this properly on iOS (simulator + iPhone X) but not on android (only emulator, no device.) Although emulator seemed fine.

# Unresolved questions / Out of scope

As mentioned in the issue and linked issues, on very tiny screens regardless of if the content is scrollable or not, it's still difficult to see things. The menu is too fat!

# Reviewer checklist

This is a suggested checklist of questions reviewers might ask during their review:
